src/kafka/inflight_activation_batcher.rs

```diff
@@ -127,12 +123,40 @@ impl Reducer for InflightActivationBatcher {
     }
 
     async fn flush(&mut self) -> Result<Option<Self::Output>, anyhow::Error> {
-        if self.batch.is_empty() {
+        if self.batch.is_empty() && self.forward_batch.is_empty() {
             return Ok(None);
         }
 
-        metrics::histogram!("consumer.batch_rows").record(self.batch.len() as f64);
-        metrics::histogram!("consumer.batch_bytes").record(self.batch_size as f64);
+        let runtime_config = self.runtime_config_manager.read().await;
+
+        if !self.batch.is_empty() {
+            metrics::histogram!("consumer.batch_rows").record(self.batch.len() as f64);
+            metrics::histogram!("consumer.batch_bytes").record(self.batch_size as f64);
+        }
+
+        // Send all forward batch in parallel
+        if !self.forward_batch.is_empty() {
+            // The default demoted topic to forward tasks to is config.kafka_long_topic if not set in runtime config.
+            let topic = runtime_config
+                .demoted_topic
+                .clone()
+                .unwrap_or(self.config.kafka_long_topic.clone());
+            let sends = self.forward_batch.iter().map(|payload| {
+                self.producer.send(
+                    FutureRecord::<(), Vec<u8>>::to(&topic).payload(payload),
+                    Timeout::After(Duration::from_millis(self.config.send_timeout_ms)),
+                )
+            });
```

Comment:
One concern I have with demoted namespaces, is that the producer cluster won't always have a long topic deployed on it. For example, if a broker is connected to `kafka-events` and we need to demote to the long/limited topic, that topic only exists on `kafka-small`.  The current logic will result in the broker failing to produce.

I think we would need to expand the application configuration to have broker configuration for the demoted namespace flow.  That would also imply that the topic we demote to is also fixed and could be removed from runtime config.